source/base/optional.h

```diff
@@ -0,0 +1,131 @@
+/**
+ *  @note This file is part of Empirical, https://github.com/devosoft/Empirical
+ *  @copyright Copyright (C) Michigan State University, MIT Software license; see doc/LICENSE.md
+ *  @date 2020.
+ *
+ *  @file optional.h
+ *  @brief Audited implementation of std::optional.
+ *  @note Status: RELEASE
+ *
+ *  In release mode, functions identically to std::optional.
+ *  In debug mode, operator * and operator-> value accesses are
+ *  checked for undefined behavior.
+ */
+
+#ifndef EMP_OPTIONAL_H
+#define EMP_OPTIONAL_H
+
+#include <type_traits>
+#include <utility>
+
+#include "assert.h"
+
+#ifdef EMP_NDEBUG
+
+namespace emp {
+  template <typename T> using optional = std::optional<T>;
+
+  template <class T>
+  inline decltype(auto) make_optional(T&& val) {
```

Comment:
can std::make_optional ever return a(n r- or l-value) reference?

aka do we need `decltype(auto)`'s perfect forwarding here, or can we just return `auto` (which just returns by value)?

Reply:
I don't /think/ you can just return auto in c++17 (& decltype(auto) is some kind of wacky workaround that restriction) but I'll check and see if it compiles

Reply:
huh, you're totally right... it did just let me use `auto`, fixed :sparkles: